packages/flutter/lib/src/widgets/single_child_scroll_view.dart

```diff
@@ -588,13 +588,13 @@ class _RenderSingleChildViewport extends RenderBox with RenderObjectWithChildMix
   void showOnScreen([RenderObject child]) {
     // Logic duplicated in [RenderViewportBase.showOnScreen].
     if (child != null) {
-      // Move viewport the smallest distance to bring [child] on screen.
       final double leadingEdgeOffset = getOffsetToReveal(child, 0.0);
       final double trailingEdgeOffset = getOffsetToReveal(child, 1.0);
       final double currentOffset = offset.pixels;
-      if ((currentOffset - leadingEdgeOffset).abs() < (currentOffset - trailingEdgeOffset).abs()) {
+
+      if (currentOffset > leadingEdgeOffset && currentOffset > trailingEdgeOffset) {
         offset.jumpTo(leadingEdgeOffset);
-      } else {
+      } else if (currentOffset < leadingEdgeOffset && currentOffset < trailingEdgeOffset ) {
         offset.jumpTo(trailingEdgeOffset);
```

Comment:
any chance we can factor the common code into the base class?

Reply:
I've looked into doing that before and there is no good common ancestor for this...

Reply:
create a common mixin (`RenderViewportShowOnScreenMixin`) or make a static method on some class (`RenderViewportBase.showChildInViewport` or some such) and use that.

Reply:
Mixin doesn't work (or gets awkward because of the super call in the method. I changed it to a static method.